### DISCUSSION TITLE
`qsv search --select` for a column with a dash?

### DISCUSSION THREAD

**Discussion by rzmk** [dathere/qsv#2098] [Q&A] [answered]
Curious how to do it. For example given:

```csv
id,item-thing,price
0,wallet,9.99
1,comb,1.39
2,pencil,0.49
```

What would be the working command to run a literal search on the `item-thing` column for `comb`?

**Comment by aborruso** [comment, marked as answer]
Hi @rzmk, two ways among many:

```
qsv search -s /item-thing/ "comb" input.csv
qsv search -s 2 "comb" input.csv
```


There is a good documentation.

```
qsv search --help
```

![image](https://github.com/user-attachments/assets/53cc3c9d-3b64-4afa-9f7c-3f027198227c)

Basically it's `qsv search [options] <regex> [<input>]`

In the option you have `-s`: Select the columns to search. See 'qsv select -h' for the full syntax.

And once again a good documentation for select. If you run

`qsv select -h`

you get several examples

```
  Select the first and fourth columns:
  $ qsv select 1,4

  Select the first 4 columns (by index and by name):
  $ qsv select 1-4
  $ qsv select Header1-Header4

  Ignore the first 2 columns (by range and by omission):
  $ qsv select 3-
  $ qsv select '!1-2'

  Select the third column named 'Foo':
  $ qsv select 'Foo[2]'

  Select columns using a regex using '/<regex>/':
  $ qsv select /^a/
  $ qsv select '/^.*\d.*$/'
  # remove SSN, account_no and password columns
  $ qsv select '!/SSN|account_no|password/'
```

**Reply by rzmk** [reply]
Thanks for the detailed explanation! I didn't realize that I could try surrounding the column names with `/`.
